### PR TITLE
Fix 10182: Set SafeBrowsing endpoint at build time

### PR DESCRIPTION
**Pull request by jumde**
Resolves: #10182 

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [ ] If adding a new directory with NPM packages, ensure this directory is
  added to `scripts/audit.js`.
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection

## Test Plan:

For developers: Before building please use the npmconfig specified here - https://github.com/brave/devops/wiki/npm-config-for-Brave-Developers

1. Open Brave Browser with a clean profile
2. Wait for ~5 mins  and confirm `<user-data-dir>/Safe Browsing` has the `*.store` files
3. Using Little Snitch or fiddler ensure that we are not reaching out to googleapis.com
4. Navigate to `https://testsafebrowsing.appspot.com` - Confirm that webpage tests work.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions.

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -141,7 +141,7 @@ def startBraveBrowserBuild() {
                                 credentials('brave-builds-github-token-for-pr-builder')
                                 github('brave/devops', 'https')
                             }
-                            branch('master')
+                            branch('mplesa-jenkins-safe-browsing-endpoint')
```
Comment (jumde) [inline Jenkinsfile:144]: Will be reverted once binaries are available.

File: README.md [hunk 2]
```diff
@@ -187,6 +187,10 @@ brave-browser> npm run apply_patches
 ...Applying 2 patches...
 ```
 
+# Enabling third party APIs:
+
+1. Google Safe Browsing: Get an API Key with SafeBrowsing API enabled from https://console.developers.google.com/ and update the value of `google_api_key` in `lib/config.js` with your key.
```
Comment (fmarier) [inline README.md:192]: How about:
> update the value of `google_api_key` in your `lib/config.js` with your key or define the `GOOGLE_API_KEY` environment variable as per <https://www.chromium.org/developers/how-tos/api-keys>.

File: lib/config.js [hunk 3]
```diff
@@ -148,7 +147,9 @@ Config.prototype.buildArgs = function () {
     is_debug: this.isDebug(),
     dcheck_always_on: this.isDcheckAlwaysOn(),
     brave_channel: this.channel,
-    google_api_key: this.googleApiKey,
+    // Replace the 'dummytoken' with your API key to use SafeBrowsing APIs
```
Comment (fmarier) [inline lib/config.js:150]: I would suggest making it more general:
> to use Google APIs such as Safe Browsing

File: README.md [hunk 4]
```diff
@@ -187,6 +187,10 @@ brave-browser> npm run apply_patches
 ...Applying 2 patches...
 ```
 
+# Enabling third party APIs:
```
Comment (fmarier) [inline README.md:190]: nit: "third-party"

File: lib/test.js [hunk 5]
```diff
@@ -56,7 +56,9 @@ const test = (suite, buildConfig = config.defaultBuildConfig, options) => {
     braveArgs.push('--test-launcher-jobs=' + options.test_launcher_jobs)
   }
 
+  const args = util.buildArgsToString(config.buildArgs())
   // Build the tests
+  util.run('gn', ['gen', config.outputDir, '--args="' + args + '"'], config.defaultOptions)
```
Comment (bridiver) [inline lib/test.js:61]: this is wrong and should be reverted

Comment (bridiver) [reply lib/test.js:61]: this will wipe out any changes passed on the command line to build and potentially cause unexpected large rebuilds

**Pull request merged**